pkg/awsvpc/parser.go

```diff
@@ -0,0 +1,344 @@
+	*vpcmodel.MultipleVPCConfigs, error) {
+	res := vpcmodel.NewMultipleVPCConfigs("AWS Cloud") // map from VPC UID to its config
+	var err error
+
+	// map to filter resources, if certain VPC, resource-group or region list to analyze is specified,
+	// skip resources configured outside that VPC
+	shouldSkipVpcIds := rc.filterByVpc(vpcID)
+
+	err = rc.getVPCconfig(res, shouldSkipVpcIds)
+	if err != nil {
+		return nil, err
+	}
+
+	var vpcInternalAddressRange map[string]*ipblock.IPBlock // map from vpc name to its internal address range
+
+	subnetNameToNetIntf := map[string][]*commonvpc.NetworkInterface{}
+	netIntfToSGs := map[string][]types.GroupIdentifier{}
+	err = rc.getInstancesConfig(subnetNameToNetIntf, netIntfToSGs, res, shouldSkipVpcIds)
+	if err != nil {
+		return nil, err
+	}
+	vpcInternalAddressRange, err = rc.getSubnetsConfig(res, subnetNameToNetIntf, shouldSkipVpcIds)
+	if err != nil {
+		return nil, err
+	}
+	// assign to each vpc object its internal address range, as inferred from its subnets
+	err = commonvpc.UpdateVPCSAddressRanges(vpcInternalAddressRange, res)
+	if err != nil {
+		return nil, err
+	}
+
+	err = rc.getSGconfig(res, shouldSkipVpcIds, netIntfToSGs)
+	if err != nil {
+		return nil, err
+	}
+
+	printVPCConfigs(res)
+
+	return res, nil
+}
+
+func (rc *AWSresourcesContainer) getVPCconfig(
+	res *vpcmodel.MultipleVPCConfigs,
+	skipByVPC map[string]bool) error {
+	for _, vpc := range rc.VpcsList {
+		if skipByVPC[*vpc.VpcId] {
+			continue // skip vpc not specified to analyze
+		}
+
+		vpcNodeSet, err := commonvpc.NewVPC(*vpc.VpcId, *vpc.VpcId, "", nil, nil)
+		if err != nil {
+			return err
+		}
+
+		newVPCConfig := commonvpc.NewEmptyVPCConfig()
+		newVPCConfig.UIDToResource[vpcNodeSet.ResourceUID] = vpcNodeSet
+		newVPCConfig.VPC = vpcNodeSet
+		res.SetConfig(vpcNodeSet.ResourceUID, newVPCConfig)
+	}
+	if len(res.Configs()) == 0 {
+		return errors.New("could not find any VPC to analyze")
+	}
+	return nil
+}
+
+func (rc *AWSresourcesContainer) getInstancesConfig(
+	subnetIDToNetIntf map[string][]*commonvpc.NetworkInterface,
+	netIntfToSGs map[string][]types.GroupIdentifier,
+	res *vpcmodel.MultipleVPCConfigs,
+	skipByVPC map[string]bool,
+) error {
+	for _, instance := range rc.InstancesList {
+		vpcUID := *instance.VpcId
+		if skipByVPC[vpcUID] {
+			continue
+		}
+		vpc, err := commonvpc.GetVPCObjectByUID(res, vpcUID)
+		if err != nil {
+			return err
+		}
+		vsiNode, err := commonvpc.NewVsi(*instance.InstanceId, *instance.InstanceId, *instance.Placement.AvailabilityZone, vpc, res)
+		if err != nil {
+			return err
+		}
+		vpcConfig := res.Config(vpcUID)
+		vpcConfig.NodeSets = append(vpcConfig.NodeSets, vsiNode)
+		vpcConfig.UIDToResource[vsiNode.ResourceUID] = vsiNode
+		for j := range instance.NetworkInterfaces {
+			netintf := instance.NetworkInterfaces[j]
+			// netintf has no CRN, thus using its ID for ResourceUID
```

Comment:
Above it says that `VPCConfigsFromResources returns a map from VPC UID (string)`; so is UID used or CRN? in fact, I don't think I've seen CRN mentioned before this place - thus the specific mentioning CRN here is very confusing

Reply:
so you removed the comment which was not relevant?

Reply:
yes
in ibm we use crn (but in netintf we use uid because it does not have crn) in aws we use uid all the time